test/conv_equals.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2023 Joachim Breitner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joachim Breitner
+-/
+
+import Std.Tactic.Basic
+import Std.Tactic.GuardMsgs
+
+
+-- The example from the doc string
+
+/-- warning: declaration uses 'sorry' -/
+#guard_msgs in
+example (P : (Nat → Nat) → Prop) : P (fun n ↦ n - n) := by
+  conv in (_ - _) => equals 0 =>
+    -- current goal: ⊢ n - n = 0
```

Comment:
You might need to import `Std.Tactic.GuardExpr`, but then you can do
```suggestion
    guard_target = n - n = 0
```
and before your sorry, `guard_target = P (fun n => 0)`

Reply:
Added! I left the comments in so that one can easily copy the test into the docstring.